package.json

```
@@ -2,20 +2,16 @@
   "name": "wiki-saikou",
   "version": "3.3.1",
   "description": "The library provides the out of box accessing to MediaWiki API in both browsers & Node.js, and the syntax is very similar to vanilla `new mw.Api()`. TypeScript definition included~",
-  "type": "commonjs",
-  "main": "./lib/index.js",
-  "types": "./lib/index.d.ts",
-  "browser": "./dist/index.umd.js",
-  "module": "./dist/index.mjs",
+  "type": "module",
+  "main": "./dist/index.cjs",
+  "module": "./dist/index.js",
+  "types": "./dist/index.d.ts",
+  "browser": "./dist/index.umd.cjs",
   "exports": {
     ".": {
-      "types": {
-        "import": "./lib/index.d.ts",
-        "require": "./lib/index.d.ts"
-      },
-      "import": "./dist/index.mjs",
-      "require": "./lib/index.js",
-      "browser": "./dist/index.umd.js"
+      "import": "./dist/index.js",
+      "require": "./dist/index.cjs",
+      "types": "./dist/index.d.ts"
```

```suggestion
      "types": {
        "import": "./dist/index.d.ts",
        "require": "./dist/index.cjs"
      },
      "import": "./dist/index.js",
      "require": "./dist/index.cjs",
      "browser": "./dist/index.umd.cjs"
```

1. Due to the limit of TypeScript, the `"types"` field should be the first entry of `exports`. 由于 TypeScript 的局限性，`"types"` 字段应该写在最前面。
2. When `moduleResolution` set to `Bundler`, you'll need an extra `.d.cts` file for CommonJS export. `moduleResolution` 设置成 `Bundler` 的时候应该添加一个额外的 `.d.cts` 给 CommonJS 用（文件内容可以和 `.d.ts` 相同，见下方的 `vite.config.ts` 文件改动
3. Add this line in `vite.config.ts` file to generate the `.d.cts` file. 把这几行加到 `vite.config.ts` 文件就可以生成 `.d.cts` 文件了。

```ts
  plugins: [
    dts(),
    {
      name: 'cts',
      closeBundle() {
        cpSync(resolve(import.meta.dirname, 'dist/index.d.ts'), resolve(import.meta.dirname, 'dist/index.d.cts'))
      }
    },
  ],
```